eli5/permutation_importance.py

```diff
@@ -0,0 +1,61 @@
+"""
+A module for computing feature importances by measuring how score decreases
+when a feature is not available.
+
+A similar method is described in Breiman, "Random Forests", Machine Learning,
+45(1), 5-32, 2001 (available online at
+https://www.stat.berkeley.edu/%7Ebreiman/randomforest2001.pdf), with an
+application to random forests. It is known in literature as
+"Mean Decrease Accuracy (MDA)" or "permutation importance".
+"""
+from __future__ import absolute_import
+
+import numpy as np  # type: ignore
+from sklearn.utils import check_random_state  # type: ignore
+
+
+def iter_shuffled(X, columns_to_shuffle=None, pre_shuffle=False,
```

Comment:
In `PermutationImportances` n_iter still works fine if you change this to pre_shuffle, as X is pre-shuffled differently for each iteration. It is a performance optimization which shouldn't affect the result if one just uses all columns once. There were two reasons to implement this feature:

1) pre-shuffling was faster for some reason when a number of columns get larger, probably because of Python overhead;
2) I was thinking it could be useful when we start to shuffle several columns at the same time.

Actually I implemented pre-shuffling first, but then switched to on-demand shuffling, as based on benchmarks it seems faster in common cases, and requires less memory. But in some cases pre-shuffling is faster, so I just kept it.